sklearn/metrics/cluster/tests/test_supervised.py

```diff
@@ -456,9 +458,9 @@ def test_adjusted_rand_score_overflow():
     rng = np.random.RandomState(0)
     y_true = rng.randint(0, 2, 100_000, dtype=np.int8)
     y_pred = rng.randint(0, 2, 100_000, dtype=np.int8)
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
```

Comment:
According to the issue linked in this test, the warning was a RuntimeWarning for overflow
```suggestion
        warnings.simplefilter("error", RuntimeWarning)
```

Reply:
Yes, I found it after I made the commit yesterday, pushing the fix now